spyder/plugins/workingdirectory/container.py

```diff
@@ -0,0 +1,301 @@
+                "created in the editor"
+            )
+        )
+        self.pathedit.setMaxCount(self.get_option('working_dir_history'))
+        self.pathedit.selected_text = self.pathedit.currentText()
+
+        # Signals
+        self.pathedit.open_dir.connect(self.chdir)
+        self.pathedit.activated[str].connect(self.chdir)
+
+    # --- PluginWidget API
+    # ------------------------------------------------------------------------
+    def setup(self, options=DEFAULT_OPTIONS):
+        self.previous_action = self.create_action(
+            WorkingDirectoryActions.Previous,
+            text=_('Back'),
+            tip=_('Back'),
+            icon=self.create_icon('previous'),
+            triggered=self.previous_directory,
+        )
+        self.next_action = self.create_action(
+            WorkingDirectoryActions.Next,
+            text=_('Next'),
+            tip=_('Next'),
+            icon=self.create_icon('next'),
+            triggered=self.next_directory,
+        )
+        browse_action = self.create_action(
+            WorkingDirectoryActions.Browse,
+            text=_('Browse a working directory'),
+            tip=_('Browse a working directory'),
+            icon=self.create_icon('DirOpenIcon'),
+            triggered=self.select_directory,
+        )
+        parent_action = self.create_action(
+            WorkingDirectoryActions.Parent,
+            text=_('Change to parent directory'),
+            tip=_('Change to parent directory'),
+            icon=self.create_icon('up'),
+            triggered=self.parent_directory,
+        )
+
+        for item in [self.previous_action, self.next_action, self.pathedit,
+                     browse_action, parent_action]:
+            self.add_item_to_toolbar(
+                item,
+                self.toolbar,
+                section=WorkingDirectoryToolBarSections.Main,
+            )
+
+    def update_actions(self):
+        self.previous_action.setEnabled(
+            self.histindex is not None and self.histindex > 0)
+        self.next_action.setEnabled(
+            self.histindex is not None
+            and self.histindex < len(self.history) - 1
+        )
+
+    def on_option_update(self, option, value):
+        if option == 'history':
+            self.history = value
+
+    # --- API
+    # ------------------------------------------------------------------------
+    def get_workdir(self):
+        """
+        Get current workdir from the file.
+
+        Returns
+        -------
+        str:
+            The current working directory.
+        """
+        if self.get_option('startup/use_fixed_directory'):
+            workdir = self.get_option('startup/fixed_directory')
+        elif self.get_option('console/use_project_or_home_directory'):
+            workdir = get_home_dir()
+        else:
+            workdir = self.get_option('console/fixed_directory')
+
+        if not os.path.isdir(workdir):

```

Comment:
Please use `osp` instead of `os.path` in this PR. That's a convention used throughout our entire codebase and we've been using it for years, so I don't agree with changing it.

Reply:
Fixed